Reorganize test data

Bai-Li-NOAA opened the pull request:
<!---
Thanks for opening a PR. This commented text will **NOT** appear in the final PR. Toggle between Write and Preview to see what your PR will look like without the comments.
-->

# What is the feature?
* This PR reorganizes test data and addresses issue #730 

# How have you implemented the solution?
* Use test fixtures to properly generate multiple combinations of available data
* Add integration tests to validate handling of missing values in the datasets

# Does the PR impact any other area of the project, maybe another repo?
* Other branches will need to be rebased with updated tests.

kellijohnson-NOAA commented on tests/testthat/test-integration-fims-estimation-with-wrappers.R:
```diff
@@ -1,8 +1,43 @@
+om_output <- om_output_list[[iter_id]]
+em_input <- em_input_list[[iter_id]]
+
+# Define modified parameters for different modules
```
In this list that you are creating, why are you using `om_input_list[[1]]`, `om_output_list[[1]]`, and `em_input_list[[1]]` instead of the objects created on lines 8--10?

Bai-Li-NOAA replied:
Great catch! I’ve reviewed the codebase and updated all relevant instances of [[1]] to [[iter_id]].

kellijohnson-NOAA commented on tests/testthat/test-integration-fims-estimation-with-wrappers.R:
```diff
@@ -1,8 +1,43 @@
+  ),
+  survey1 = list(
+    LogisticSelectivity.inflection_point.value = 1.5,
+    LogisticSelectivity.slope.value = 2,
```
Isn't 2 the default? If yes, then do we need to specify the default here?

kellijohnson-NOAA commented on tests/testthat/test-integration-fims-estimation-with-wrappers.R:
```diff
@@ -257,183 +290,166 @@ test_that("estimation test of fims using wrapper functions", {
-    data = fims_data
+  # Compare FIMS results with model comparison project OM values
+  validate_fims(
+    report = agecomp_fit@report,
```
For this line and the subsequent line we should be using `get_report()` and `get_estimates()`.

kellijohnson-NOAA commented on tests/testthat/test-integration-fims-estimation-with-wrappers.R:
```diff
@@ -257,183 +290,166 @@ test_that("estimation test of fims using wrapper functions", {
+  validate_fims(
+    report = agecomp_fit@report,
+    sdr = agecomp_fit@estimates,
+    sdr_report = agecomp_fit@estimates,
```
How come the sdr_report is the estimates table?

Bai-Li-NOAA replied:
I’ve updated the tests to use get_report() and get_estimates() when working with the wrapper functions.

kellijohnson-NOAA commented on tests/testthat/test-integration-fims-estimation-with-wrappers.R:
```diff
@@ -257,183 +290,166 @@ test_that("estimation test of fims using wrapper functions", {
-    report = fit@report,
-    sdr = fit@estimates,
-    sdr_report = fit@estimates,
+    report = agecomp_na_fit@report,
+    sdr = agecomp_na_fit@estimates,
+    sdr_report = agecomp_na_fit@estimates,
```
Same comments as above.

Bai-Li-NOAA replied:
Done.

kellijohnson-NOAA commented on tests/testthat/test-integration-fims-estimation-with-wrappers.R:
```diff
@@ -257,183 +290,166 @@ test_that("estimation test of fims using wrapper functions", {
-    )
+  # Compare FIMS results with model comparison project OM values
+  validate_fims(
+    report = lengthcomp_fit@report,
+    sdr = lengthcomp_fit@estimates,
+    sdr_report = lengthcomp_fit@estimates,
```
Same comments as above.

Bai-Li-NOAA replied:
Done

kellijohnson-NOAA commented on tests/testthat/test-integration-fims-estimation-with-wrappers.R:
```diff
@@ -257,183 +290,166 @@ test_that("estimation test of fims using wrapper functions", {
+
+  # Compare FIMS results with model comparison project OM values
+  validate_fims(
+    report = lengthcomp_na_fit@report,
+    sdr = lengthcomp_na_fit@estimates,
+    sdr_report = lengthcomp_na_fit@estimates,
```
Same as above.

Bai-Li-NOAA replied:
Done

kellijohnson-NOAA commented on tests/testthat/test-integration-fims-estimation-with-wrappers.R:
```diff
@@ -257,183 +290,166 @@ test_that("estimation test of fims using wrapper functions", {
-    report = fit@report,
-    sdr = fit@estimates,
-    sdr_report = fit@estimates,
+    report = age_length_comp_na_fit@report,
+    sdr = age_length_comp_na_fit@estimates,
+    sdr_report = age_length_comp_na_fit@estimates,
```
Same as above.

Bai-Li-NOAA replied:
Done

kellijohnson-NOAA reviewed (commented):
Thanks @Bai-Li-NOAA for these changes. I think this work is a great movement towards cleaning up the codebase as a whole and making things more consistent. I added two commits that should be squashed when you are ready. I do have a question about the sdr_report and the use of @.

Bai-Li-NOAA commented:
@kellijohnson-NOAA, I have addressed the great suggestions! I wanted to note a few additional changes that may need further review:
- I've updated `.vscode/settings.json` to include editor rulers at 80 characters. This will display a vertical line in VS Code automatically for developers.
- I’ve refactored `tests/testthat/helper-integration-tests-validation.R` to reduce duplicate code, making it approximately half as long.
- @Andrea-Havron-NOAA discovered that the tests aren’t estimating recruitment log_devs. A few tests will fail if we estimate log_devs, so I’ve added #TODO tasks in the code to remind us to estimate log_devs once the issue is resolved.

kellijohnson-NOAA commented:
@Bai-Li-NOAA great job. Please rebase and merge into dev whenever you see fit.
